sdk/python/kfp/components/pipeline_task.py

```diff
@@ -474,6 +475,42 @@ def my_pipeline():
             self._task_spec.dependent_tasks.append(task.name)
         return self
 
+    def ignore_upstream_failure(self) -> 'PipelineTask':
+        """If called, the pipeline task will run when any specified upstream
+        tasks complete, even if unsuccessful.
+
+        This method effectively turns the caller task into an exit task
+        if the caller task has upstream dependencies.
+
+        "If the task has no upstream tasks, either via data exchange or an explicit dependency via .after(), this method has no effect."
+
+        Returns:
+            Self return to allow chained setting calls.
+
+        Example:
+          ::
+
+            @dsl.pipeline()
+            def my_pipeline(sample_input1: str = 'message'):
+                task = fail_op(message=sample_input1)
+                clean_up_task = print_op(
+                    message=task.output).ignore_upstream_failure()
+        """
+
+        for input_spec_name, input_spec in (self.component_spec.inputs or
+                                            {}).items():
+            argument_value = self._inputs[input_spec_name]
+            if (isinstance(argument_value, pipeline_channel.PipelineChannel)
+               ) and (not input_spec.optional) and (argument_value.task_name
+                                                    is not None):
+                raise ValueError(
+                    f' Task {self.name} requires a default value to make sure it never fails.'
```

Comment:
It can be useful to construct error messages first with what the requirement is, then what specific user action didn't adhere to the requirement. WDYT about the following?

`f'Tasks can only use .ignore_upstream_failure() if all input parameters that accept arguments created by an upstream task have a default value, in case the upstream task fails to produce its output. Input parameter task {self.name}'s {input_spec_name} argument is an output of an upstream task {argument_value.task_name}, but {input_spec_name} has no default value.'`

Perhaps there's a less verbose way to put this. Just want to provide a little more instruction for the user.

You can also use `!r` to normalize formatting of the interpolated strings, similar to [this](https://github.com/kubeflow/pipelines/blob/a45d72d75dd99c9cbb8dc4706d380e4ceb9ec47e/sdk/python/kfp/components/pipeline_task.py#L319).